### ISSUE TITLE
Convert to TypeScript?

### ISSUE THREAD

**Issue by wbern** [hbatalhaStch/react-big-scheduler#12]
### Checklist

- [X] I've looked through the [README](https://github.com/hbatalhaStch/react-big-scheduler#readme)
- [X] I've verified that I'm running react-big-scheduler-stch version **1.2.2**

### Please make sure the question is worded well enough to be understood

Maybe it's time to consider converting the project to TypeScript? It will make it easier to debug issues on the user end.

There are CLI's we can use that converts everything but adds @ts-expect-error [error message] for every error.

I can supply a PR if it sounds interesting?

**Comment by hbatalhaStch** [comment]
Yeah I was thinking about the idea of converting the project to TS after I merged your PR.
>There are CLI's we can use that converts everything but adds @ts-expect-error [error message] for every error.

Do you have any CLI in particular in mind?

>I can supply a PR if it sounds interesting?

I have never converted a project from JS to TS so feel free to lead the way

**Comment by diragb** [comment]
I'd be happy to pick this up and work on it! 🙂
Also - I only get time to work on side-projects on weekends, so I might take some time to finish this up. If anyone is actively working towards this and has more time to contribute, please let me know here or via email at **hey@diragb.dev**.

**Comment by hbatalhaStch** [comment]
> I'd be happy to pick this up and work on it! 🙂 Also - I only get time to work on side-projects on weekends, so I might take some time to finish this up. If anyone is actively working towards this and has more time to contribute, please let me know here or via email at **[hey@diragb.dev](mailto:hey@diragb.dev)**.

Thank you for taking up this, it is a huge help. We could do this together, actively pushing commits towards converting this project to TypeScript. Is there any particular way you want  to this? Let me know
